aws_lambda_powertools/utilities/idempotency/idempotency.py

```diff
@@ -132,6 +133,8 @@ def handle(self) -> Any:
             # We call save_inprogress first as an optimization for the most common case where no idempotent record
             # already exists. If it succeeds, there's no need to call get_record.
             self.persistence_store.save_inprogress(event=self.event, context=self.context)
+        except IdempotencyKeyError as e:
+            raise e
```

Comment:
End of the day for me so I need to test as I touched many subjects today... 

Because we're simply re-raising, you can do:

```suggestion
        except IdempotencyKeyError:
            raise
```

Reply:
Got to love the Pythonic way :)